decouple.py

```diff
@@ -147,13 +147,21 @@ def get(self, key):
 class AutoConfig(object):
     """
     Autodetects the config file and type.
+
+    Parameters
+    ----------
+    search_path : str, optional
+        Initial search path. If empty, the default search path is the
+        caller's path.
+
     """
     SUPPORTED = {
         'settings.ini': RepositoryIni,
         '.env': RepositoryEnv,
     }
 
-    def __init__(self):
+    def __init__(self, search_path=None):
+        self.search_path = search_path
```

Comment:
If `search_path` is a relative path like `.`, when ipython does the `exec` the result will a path relative to wherever ipython executes the code.

We could either:
- Make a note expecting the user to pass an absolute path.
- If search_path is given, make it absolute.